packages/components/src/DataList/DataList.utils.tsx

```diff
@@ -137,3 +165,171 @@ export function generateDataListEmptyState({
 
   return EmptyStateComponent;
 }
+
+/**
+ * Render the data list items with the layout that
+ * should be visible based on its size prop and the size props of the other layouts
+ */
+export function renderDataListItems<T extends DataListObject>(
+  layouts: React.ReactElement<DataListLayoutProps<T>>[] | undefined,
+  elementData: DataListItemType<T[]>[],
+  mediaMatches?: Record<Breakpoints, boolean>,
+) {
+  return renderDataListLayouts(
+    layouts,
+    (layout: React.ReactElement<DataListLayoutProps<T>>) => {
+      return elementData.map((child, i) => {
+        // TODO: Don't use index as key. Might have to force an ID on the data JOB-76773
+        return (
+          <div className={styles.listItem} key={`${i}-${child.id}`}>
```

Comment:
hmmm. that key isn't gonna work since it is still tied to the index and therefore would rerender the moment that index changes when another item gets inserted.

Reply:
see PR description https://github.com/GetJobber/atlantis/pull/1496